.github/workflows/gh-container-image.yml

```diff
@@ -0,0 +1,36 @@
+name: Update container image
+
+on:   
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
Why is this needed? It will build image on every PR against `main`. `push` above should be sufficient.